src/bar.cpp

```diff
@@ -665,7 +665,11 @@ void waybar::Bar::onMap(GdkEventAny*) {
 
 void waybar::Bar::setVisible(bool value) {
   visible = value;
-  setMode(visible ? MODE_DEFAULT : MODE_INVISIBLE);
+  if (auto mode = config.get("mode", {}); mode.isString()) {
+    setMode(visible ? config["mode"].asString() : MODE_INVISIBLE);
```

Comment:
Do you think it's worse to check if mode is available?
Against a hard-coded list, for example.

Reply:
`setMode` already does the check with fallback to default, no need to do it twice.